lib/linux_kernel/kthread.c

```diff
@@ -228,20 +228,24 @@ M0_INTERNAL int m0_thread_confine(struct m0_thread *q,
 		  put_task_struct(p);
 		*/
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
+#ifdef CONFIG_X86_64
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0)
 		cpumask_copy(&p->cpus_mask, cpuset);
-#else
+	#else
 		cpumask_copy(&p->cpus_allowed, cpuset);
-#endif
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
+	#endif
+	#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
 		p->nr_cpus_allowed = nr_allowed;
-#else
+	#else
 		p->rt.nr_cpus_allowed = nr_allowed;
+	#endif
+#else /*aarch64*/
+	cpumask_copy(&p->cpus_allowed, cpuset);
+	p->nr_cpus_allowed = nr_allowed;
 #endif
```

Comment:
Why is this change needed?

1st, it is wrong. It does not check kernel version for aarch64 architectures. 2nd, as CentOS-7 uses kernel version 4.18 for aarch64 - the old code should work fine on aarch64. Let me know what I'm missing.

Reply:
The explicit checking for Linux kernel version is not required for ARM platform unlike in X86_64 platform since ARM platform only support 4.18 and on words.  The above code for x86_64 would not works for ARM platform because the 4.18 version of Linux is not compatible  in all respect in both the platform(X86_64 , AARCH64). The set of code specific to ARM platform are weaved out and placed in its own block however.